exercises/drill-enum1-replace-nils.livemd

```diff
@@ -0,0 +1,163 @@
+# Drill: Replace `nil`s in a List with Elements from a 2nd List
+
+```elixir
+Mix.install([
+  {:youtube, github: "brooklinjazz/youtube"},
+  {:hidden_cell, github: "brooklinjazz/hidden_cell"},
+  {:tested_cell, github: "brooklinjazz/tested_cell"},
+  {:utils, path: "#{__DIR__}/../utils"}
+])
+```
+
+## Navigation
+
+[Return Home](../start.livemd)<span style="padding: 0 30px"></span>
+[Report An Issue](https://github.com/DockYard-Academy/beta_curriculum/issues/new?assignees=&labels=&template=issue.md&title=)
+
+## Purpose
+
+This is a *drill* exercise:
+
+> *Drill* exercises are meant to provide practise of Elixir's
+> syntax and important language modules so developers can
+> type them out as fast as possible.
+> The problem should not be conceptually difficult to
+> facilitate this goal.
+> 
+> Developers need to commit both Elixir's syntax and core
+> modules/functions to (working) memory so that when writing
+> their code the developer is not slowed down having to
+> remember the syntax or which module function to use. This
+> is analogous to knowing how to spell and the grammar of
+> your (human) language.
+
+In this particular iteration of this problem, the goal is
+to practise writing/using functions from the `Enum` module;
+some important ones being:
+
+* Iteration: `map`, `filter`, `reduce`, `zip`
+* Ordering: `reverse`, `sort`, `with_index`, `shuffle`
+* Grouping: `chunk*`, `take*`, `drop*`, `frequencies`, `group_by`
+* Conditionals: `member?`, `any?`, `all?`, `find*`
+
+The `Enum` module is probably the main Elixir module and
+you should definitely work to commit as much as possible
+to memory.
+
+## Replace `nil`s in a List with Elements from a 2nd List
+
+Now that you know how to iterate over a list, your assignment
```

Comment:
Maybe we can simplify this description? I had a hard time following it until I saw the test cases.

Here's an example:

To develop familiarity with the `Enum` module, you will replace `nil` values in a list with values from a second list at the same index. For example:

```elixir
list1 = [0, nil, 2, 3, nil]
list2 = [:a, :b, :c, :d, :e]

ReplaceNils.replace(list1, list2)
[0, :b, 2, 3, :e]
```

Implement the `ReplaceNils` module as documented below.
...etc

Reply:
done!